registry/registryprovider.go

```diff
@@ -349,7 +349,7 @@ func ShortTypeToDownloadURLs(rp RegistryProvider, t string) ([]string, Registry,
 	}
 
 	if r == nil {
-		panic(fmt.Errorf("cannot get github registry for %s", t))
+		return nil, nil, fmt.Errorf("cannot get github registry for %s", t)
```

Comment:
There's 2 other panics in this file, can you change them as well or put a TODO there. There's also one in registry.go and deployments.go.